terragrunt/aws/api/lambda.tf

```
@@ -0,0 +1,32 @@
+module "url_shortener_lambda" {
+  source                 = "github.com/cds-snc/terraform-modules?ref=v5.0.0//lambda"
+  name                   = "${var.product_name}-api"
+  billing_tag_value      = var.billing_code
+  ecr_arn                = aws_ecr_repository.api.arn
+  enable_lambda_insights = true
+  image_uri              = "${aws_ecr_repository.api.repository_url}:latest"
+  memory                 = 3008
+  timeout                = 300
+  ephemeral_storage      = 768
+
+  vpc = {
+    security_group_ids = [aws_security_group.api.id]
+    subnet_ids         = module.url_shortener_vpc.private_subnet_ids
+  }
+
+  environment_variables = {
+    DOMAIN                    = var.domain
+    API_AUTH_TOKEN_SECRET_ARN = aws_ssm_parameter.api_auth_token
```

You'll just need to add the `arn` property to this resource:
```suggestion
    API_AUTH_TOKEN_SECRET_ARN = aws_ssm_parameter.api_auth_token.arn
```

---

So I changed things a little bit around since I realized that I should not be passing the arn to the function but instead I should be passing the api token value. I have changed it to:
`API_AUTH_TOKEN_SECRET = var.api_auth_token`
Let me know if I am totally wrong here.

---

So it would 100% work to pass in the `var.api_auth_token` to the Lambda function, but the reason we don't do it this way is because Lambda environment variables are not secure.  They get dumped as command output for a few different operations and it has actually led to an incident with us (since we did exactly this in the past).

By passing in the secret's ARN, we can retrieve the auth token value when the Lambda starts so there's no risk of accidentally leaking that value down the road.

---

Gotcha and thank you so much for the detailed explanation. I have fixed this to take the arn instead.